rlimit_posix.go

```diff
@@ -0,0 +1,24 @@
+// +build !windows
+
+package caddy
+
+import (
+	"fmt"
+	"syscall"
+)
+
+// checkFdlimit issues a warning if the OS limit for
+// max file descriptors is below a recommended minimum.
+func checkFdlimit() {
+	const min = 8192
+
+	// Warn if ulimit is too low for production sites
+	rlimit := &syscall.Rlimit{}
+	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, rlimit)
+	if err == nil {
+		if rlimit.Cur < min {
```

Comment:
I know this might seem like a nit-pic, but can we combine this to one line: `if err == nil && rlimit.Cur < min {` - and save a level of indentation? I prefer that style over separate if statements.